blips/blip-0001.md

```diff
@@ -0,0 +1,289 @@
+```
+bLIP: 1
+Title: bLIP process
+Status: Active
+Author: Ryan Gentry <ryan@lightning.engineering>
+Created: 2021-05-21
+Post-History: 2021-06-30: https://lists.linuxfoundation.org/pipermail/lightning-dev/2021-June/003086.html
+  [lightning-dev] bLIPs: A proposal for community-driven app layer and protocol extension standardization
+License: CC0
+```
+
+# Abstract
+
+bLIP stands for Bitcoin Lightning Improvement Proposal. A bLIP is a design document 
+providing information to the Lightning community, or describing a new feature for 
+the Lightning Network. The bLIP should provide a concise technical specification of 
+the feature and a rationale for the feature. The bLIP author is responsible for 
+building consensus within the community and documenting dissenting opinions.
+
+# Copyright
+
+This bLIP is licensed under the CC0 license.
+
+# Rationale
+
+As the Lightning community has grown, new features, standards, and protocols have 
+been developed outside of the BOLT specification process: particularly at the 
+application level that isn’t described within the core BOLT documents. This is great! 
+But in the spirit of interoperability, documenting features, standards, and protocols 
+in a single location with a standard format will make it easy on future adopters. 
+
+In particular, there are three scarce sets of identifiers used in Lightning Network
+protocol development that benefit from central organization and documentation to
+avoid potential clashes:
+
+* **Feature Bits** are used to designate that a given node supports a given feature, and 
+are publicly broadcasted on the Lightning Network. Feature Bits are first introduced as
+optional with *odd* bits, and when upgraded to compulsory are switched to *even* bits, 
+so a new feature proposed in a bLIP should start with an *odd* bit. Feature Bits are 
+assigned and specified in [Bolt
+ #9](https://github.com/lightningnetwork/lightning-rfc/blob/master/09-features.md).
+* **Message Types** are used to indicate how to interpret the `payload` feature of a 
+Lightning message. Types `32768`-`65535` are set aside for experimental and 
+application-specific messages, which are best suited to be documented in a bLIP. 
+Message Types are assigned and specified in [Bolt
+ #1](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md).
+* **Type-Length-Values (TLVs)** are used to allow for the backwards-compatible addition 
+of new fields to existing message types. `type` determines how the contents of `value`
+should be decoded, so specifying how to actually perform that decoding is a
+great use case for a bLIP. TLVs are not assigned anywhere, but specified in [Bolt
+ #1](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md).
```

Comment:
```suggestion
of new fields to existing message types (as described in
[Bolt #1](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md)).
bLIPs may introduce new TLV fields to existing messages,
using `type`s greater than `253`.
```

NB: I chose `253` very arbitrarily (to leave the 1-byte varints for the BOLTs), maybe we want a higher value (and we still need to specify where/how we centralize the list of such tlvs)

Reply:
The [only existing registry of TLV values](https://github.com/satoshisstream/satoshis.stream/blob/main/TLV_registry.md) I'm aware of start at 696969, so plenty of room to bump up from 253. I'll say "`type`s greater than `509`" and we can iterate from there :)